### DISCUSSION TITLE
Only works for Jandy?

### DISCUSSION THREAD

**Discussion by BlackAlgae** [aqualinkd/AqualinkD#317] [General]
Can I use aqualinkD on pentair control panels? Also, do I need to plug raspberry into a board before I install AqualinkD or can I start installing before I choose my Equipment (pentair or jandy board - still unsure) and plugin the USB2RS485. How old or new does the jandy board model have to be? I think I saw aqualinkD works with any rev after Q? I'm unsure what to buy here, my apologies.

Also had a question regarding the new pentair intelibrite architecture multi color LEDs. Can aqualinkD be used to change colors for that pool light in particular?

Many thanks! Awesome work here

**Comment by sfeakes** [comment]
AqualinkD only works for Jandy control panels.  If you want/use a Pentair panel then [nodejs pool controller](https://github.com/tagyoureit/nodejs-poolController) is similar to AqualinkD but for Pentair.

AqualinkD can work with all Jandy RS boards & versions.  IT does also support PD (or PDA Only) boards, BUT Keep away from PDA boards (they are slow and support very limited in AqualinkD).   Depending on the board version will depend on what devices it can support though.   I usually recommend Rev T2 or newer as that supports everything. [This link](https://www.poolsupplyunlimited.com/Products/Manuals/Sub53_2012329101223.pdf) will show you things that are / are not supported on different versions.

Jandy control panel does support many Pentair color LED's and Lights, but as for anything new, I'm not sure, you should talk to a Jandy rep / company.  IF the control panel supports the lights, then AqualinkD can as well.   AqualinkD can also support lights that the Jandy control panel can't, it does this by pulsing power to the light itself to set the color/mode. (This is how the majority of color lights work). But again I have not looked at the new pentair intelibrite architecture multi color LEDs.

You can run AqualinkD with just a RS485 adapter (or without one) and with a Docker you don;t need to use a Pi.  BUT, you won;t get far it needs a lot of information from the panel before it can fully startup.  So without a panel to talk to AqualinkD will just exit after a minute if it can't get a connection.

**Reply by BlackAlgae** [reply]
Thank you so much for the valuable info! 

I will definitely keep nodejs pool controller in mind. Since im getting the architecture intelibrites and jandy RS one touch for free I will most likely use the aqualinkD.

When I spoke to a Jandy rep. they said it will work with the arch pool lights but the one touch "most likely" wont let me change colors. Jandy names the colors differently. But, I will be able to use the outdoor panel to cycle through the 12 color options/modes. 

Im a little new to the scene so im not sure what a Docker is my apologies. My earlier question was, can I start installing the AqualinkD to my Pi zero w 2 now or should I wait for the RS jandy to get here. Im a litte excited about this project and would like to get a head start :D 

I will be sure to update on the arch. intelibrites once everything is installed! 

Thanks again kind sir

**Comment by sfeakes** [comment]
All Pentair lights that I've come across (and many other brands), can default to a common mode called Sam/SAL, so at the minimum I'm sure you can set the lights to that mode, and get a large variety of colors through the OneTouch remote.  It's not in the documentation, but if you look in this file, you can see all the colors I've decoded that Jandy can support for different lights.
https://github.com/sfeakes/AqualinkD/blob/master/source/color_lights.h
In that file you will see 12 Intelibrite colors, my "guess" would be that the Jandy panel could set those colors on the architecture intelibrites

For instance, I have "Color Logic" (Haywood) lights, Their are about 15 colors / modes that that light can support, and Jandy control panel supports 12 of them.  So I set AqualinkD to control the light (and not the control panel), that way I get all 15 colors.  BUT, I can only set the color through AqualinkD and not through and Jandy keypad/remote. But all remotes can turn them on / off.

Ignore the docker comment.  It's just a way to deploy software to servers.  Not relevant if you are going to use a RaspberryPi (which is by far the best / preferred way to use AqualinkD).

You can install AqualinkD on the pi to get ready, but when you start it, it will just fail after a minute or so.  So probably not much point.

The reason I posted the Pentair note js page, I (incorrectly) read your first post that you hadn't decided what control panel to get yet.  If you are getting a Jandy panel, then AqualinkD is the software you want.

**Comment by BlackAlgae** [comment]
Youre the best! Seriously

Is Sam/Sal the mode where it cycles through various colors?

 To recap, the control board lets you go through these 12 options but not the indoor control (onetouch) board? A budy of mine just sent me a video where he turns off and on the intellibrite multiplied by the number the color or mode is set to through the aux on the out door panel. For example, Usa is number 10 so he turns on and off 10 times and it goes to usa mode. He has the same jandy RS and architecture inteliB light setup.

Yes! I think i will go with the aqualinkD since im getting a aqualink RS + inteliB lights for free. Im also assuming its best to install aqualinkD while the rasbperry Pi is connected to the jandy baord via RS485?

Many many - MANY thanks again!

**Comment by sfeakes** [comment]
All color modes from different manufacturers that I have come across can be set by pulsing the light on/off (in your description 10 times). AqualinkD will do this for you, so you select the light mode you want, and it will pulse the light on/off the correct amount of times.  The Jandy control panel will do exactly the same thing, but only knows a limited number of colors, where as AqualinkD can be configured for all the colors for a specific light.
There are some lights that can be set differently (through the RS485 protocol), but I have not had the opportunity to decode / play with those. 

if you set the control panel to a specific light type (listed in the file I provided) then when you turn the light on from the one touch remote, it will give you the option to pick which color mode you want, then cycle through the correct number of pulses.  If you set the control panel to a simple light, then it will simple turn it on or off. 
In this mode you can configure AqualinkD to pulse the correct number it times for any mode you configure.

The Sam/sal mode, seems to be a “common” mode you can put many manufacturers lights into so they can be controlled by any panel (Ie the on/off pulses are standardized)

once you get the panel installed, and you can play with it, I’m sure this will be easier to understand.

**Comment by BlackAlgae** [comment]
Thats awesome! 

so aqualinkD WILL be able to choose the colors/modes for the architecture intellibrite! 

Now, on your second paragraph, are you recommending to set the pool light to a color mode -rather than a static color - so that the one touch indoor panel can access other color modes? Or buy one touch "remote" do you mean a wireless device ( separate from a indoor one touch panel that goes mounted inside the home ).  

Silly question here, once I get the jandy panel installed, should I plug in the rasbperry pi using the USB2RS485 so I can then download and install aqualinkD. In other words, I should install aqualinkD ONLY after getting this?:

^C^@JANDY AquaLinkRSp^P^C^@^P^B^@^A^@^@^S^P^C^@^P^B*^@<^P^C^P^B^H^B^@^@^@^@^@^\^P^C^@^P^B^@^A^@^@^S^P^C^P^B$
^B^@^@^@^@^@^^^P^C^@^P^B^@^A^@^@^S^P^C^@^P^B`^@r^P^C^P^B^H^B^@^@^@^@^@^\^P^C^@^P^B^@^A^@^@^S^P^C^P^B$
^B^@^@^@^@^@^^^P^C^@^P^B^@^A^@^@^S^P^C^@^P^B+^@=^P^C^P^B^H^B^@^@^@^@^@^\^P^C^@^P^B^@^A^@^@^S^P^C^P^B$
^B^@^@^@^@^@^^^P^C^@^P^B^@^A^@^@^S^P^C^@^P^B`^@r^P^C^P^B^H^B^@^@^@^@^@^\^P^C^@^P^B^@^A^@^@^S^P^C^P^B^H^C^@ AIR TEMP 87M-_F  c^P^C^@^P^B^@^A^@^@^S^P^C^P^B$
^B^@^@^@^@^@^^^P^C^@^P^B^@^A^@^@^S^P^C^@^P^B8^@J^P^C^P^B^H^B^@^@^@^@^@^\^P^C^@^P^B^@^A^@^@^S^P^C^P^B$

Lastly, you mentioned under web config this: "light_program If you have a programable light (and it's configured in aqualinkd.conf as a light for AqualinkD to program and not the control panel), then list the light modes here."

Would this be where I add the list of colors from the list you sent me? 

Since im a noob, I basically go down the config list choosing/typing the commands for what suits me and my equipment? There's no script or option that can do this for me and give me the "essentials".  I guess im lazy and nervous haha but also very excited so that wont stop me. 

Thanks again!

**Reply by sfeakes** [reply]
> Now, on your second paragraph, are you recommending to set the pool light to a color mode -rather than a static color - so that the one touch indoor panel can access other color modes? Or buy one touch "remote" do you mean a wireless device ( separate from a indoor one touch panel that goes mounted inside the home ).

This would be a preference as to how you want to control the light, and what devices you have.  Don't buy anything.
If you want all color modes, and only want to set them from AqualinkD, then use AqualinkD to configure the light color mode.
If you can get by with a limited set of colors, and want to control them from a Jandy remote & aqualinkD, then configure the panel for the color light.

> Silly question here, once I get the jandy panel installed, should I plug in the rasbperry pi using the USB2RS485 so I can then download and install aqualinkD. In other words, I should install aqualinkD ONLY after getting this?:

That doesn't mater.  You can install AqualinkD at any time you like.  What you posted is a simple test that your config is working.  you can install AqualinkD at any time, it just won't run without stay running without the RS485 connector.



> Lastly, you mentioned under web config this: "light_program If you have a programable light (and it's configured in aqualinkd.conf as a light for AqualinkD to program and not the control panel), then list the light modes here."
> 
> Would this be where I add the list of colors from the list you sent me?

That's where you configure AqualinkD to program the light. That's NOT where you list the color modes. you do that in the web UI.
Look around line 50 in this file https://github.com/sfeakes/AqualinkD/blob/master/web/config.js, that's where you place the color modes.  This is also described in the wiki.
